### ISSUE TITLE
Support NSwag easily

### ISSUE THREAD

**Issue by adrianhall** [Azure/azure-mobile-apps#253]
**Is your feature request related to a problem? Please describe.**
I want to expose the datasync controllers as an OpenAPI spec using NSwag.

**Describe the solution you'd like**
Something like:

```csharp
services.AddSwaggerDocument(config => 
{
    // Extension method that does "all the right things"
   config.AddDatasyncControllers();
});
```

**Comment by adrianhall** [comment]
Based on a discussion with Rico Suter (who wrote NSwag), I need to implement an [`IOperationProcessor`](https://github.com/RicoSuter/NSwag/blob/4f345e1adc9c49d8498d4018250760aa69dd8dd2/src/NSwag.Generation/Processors/IOperationProcessor.cs) and link it into the configuration via the extension method.  The `IOperationProcessor.Process` method takes a context.  I can do the work of figuring out the appropriate changes to make and then updating the `context.OperationDescription` and returning true to add the method to the spec.  

The `IOperationProcessor.Process` method is called on a per-method basis, and I can use Introspection to ensure it's part of a TableController<T>, plus provide the necessary work for output.

**Comment by anpin** [comment]
hi @adrianhall also it seems that this would be needed for swagger UI to send requests 
```
public class ZumoHeaderSwaggerAttribute : IOperationFilter
{
    public void Apply(OpenApiOperation operation, OperationFilterContext context)
    {
        if (operation.Parameters == null)
            operation.Parameters = new List<OpenApiParameter>();
 
        operation.Parameters.Add(new OpenApiParameter
        {
            Name = "ZUMO-API-VERSION",
            Example = new OpenApiString("2.0.0"),
            In = ParameterLocation.Header,
            Required = true,
            Schema = new OpenApiSchema
            {
                Type = "string" 
            }
        });
    }
}
```
and then 
```
builder.Services.AddSwaggerGen(options => options.OperationFilter<ZumoHeaderSwaggerAttribute>());
```

**Comment by adrianhall** [comment]
That's the trivial version.  Currently, the models and OData query arguments are not generated in the OAS document, and those are important.  Also, this makes the entire API surface subject to ZUMO=API-VERSION, which is not true (it's only on the table controllers)

Plan is to cycle through each of the table controllers and add the appropriate settings for each one.

**Comment by adrianhall** [comment]
Fixed in PR #534
